mmeds/server.py

```diff
@@ -625,28 +641,46 @@ def validate_metadata(self, myMetaData, temporary=False):
     @cp.expose
     def process_data(self, public=False, **kwargs):
         cp.log('Public is {}'.format(public))
+        
         # Create a unique dir for handling files uploaded by this user
         subject_metadata = Path(cp.session['uploaded_files']['subject'])
         specimen_metadata = Path(cp.session['uploaded_files']['specimen'])
 
         # Get the username
         username = self.get_user()
 
+        # Unpack kwargs based on barcode type
         # Add the datafiles that exist as arguments
         if cp.session['upload_type'] == 'qiime':
-            datafiles = self.load_data_files(for_reads=kwargs['for_reads'],
-                                             rev_reads=kwargs['rev_reads'],
-                                             barcodes=kwargs['barcodes'])
+            for_reads = kwargs['for_reads']
+            rev_reads = kwargs['rev_reads']
+            if not cp.session['dual_barcodes']:
+                barcodes = kwargs['barcodes']
+                reads_type = kwargs['reads_type']
+                barcodes_type = 'solo_barcodes'
+                datafiles = self.load_data_files(for_reads=for_reads,
+                                                 rev_reads=rev_reads,
+                                                 barcodes=barcodes)
+            else:
+                for_barcodes = kwargs['for_barcodes']
+                rev_barcodes = kwargs['rev_barcodes']
```

Comment:
You don't need to assign these to a local variable. Since they are only being used once it's neater to use the value directly from the dictionary like for kwargs['otu_table'] down below